### PR TITLE
Minor update to building/internals.md

### PR DESCRIPTION
**Pull request by daxian-dbw**
As discussed in https://github.com/PowerShell/PowerShell/pull/5072#issuecomment-336012983, this is a minor update to the `building/internals.md`

### CODE REVIEW

File: docs/building/internals.md [hunk 1]
```diff
@@ -59,9 +59,7 @@ you likely need to delete the `gen` folders and re-run the tool.
 
 ### Type Catalog
 
-As a work-around for the lack of `GetAssemblies()`,
-our custom assembly load context takes a pre-generated catalog of C# types
-(for PowerShell type resolution).
+A pre-generated catalog of C# types is used in PowerShell to help type resolution.
```
Comment (adityapatwardhan) [inline docs/building/internals.md]: Line 71 and below needs to be updated. CorePsTypeCatalog.ps is not under AssemblyLoadContext anymore.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: docs/building/internals.md (hunk 1 above)
Comment (daxian-dbw) [reply docs/building/internals.md]: Good catch. Will fix.

Comment (daxian-dbw) [reply docs/building/internals.md]: Oh, forgot it. That one will be addressed by #5072. That PR is pending merging.

**Review by adityapatwardhan — APPROVED** [review]

File: docs/building/internals.md (hunk 1 above)
Comment (adityapatwardhan) [reply docs/building/internals.md]: I will merge this once #5072 is merged.

Comment (daxian-dbw) [reply docs/building/internals.md]: I just merged  #5072 :)

**Pull request merged**